taichi/transforms/auto_diff.cpp

```diff
@@ -251,6 +251,105 @@ class PromoteSSA2LocalVar : public BasicStmtVisitor {
   }
 };
 
+std::set<TernaryOpType> stack_needed_ternary_collections{TernaryOpType::select};
```

Comment:
We can move these to be const static member of `AdStackAllocaJudger` class.

Reply:
Have moved.

Reply:
Have moved.